### DISCUSSION TITLE
Android APK build

### DISCUSSION THREAD

**Discussion by Torinde** [dosbox-staging/dosbox-staging#3773] [Ideas]
Pie in the sky - running Staging on phones/tablets, TVs, etc.

Preferable solution will be a regular Android app APK. It's supported by [SDL2](https://github.com/libsdl-org/SDL/blob/main/docs/README-android.md). Related [X issue](https://github.com/joncampbell123/dosbox-x/issues/959).

Alternatives considered: RetroArch ([Pure](https://github.com/dosbox-staging/dosbox-staging/issues/885), [Core](https://github.com/dosbox-staging/dosbox-staging/issues/324), #885), [Termux](https://github.com/joncampbell123/dosbox-x/issues/2658).

**Comment by johnnovak** [comment]
Short answer: No.

Detailed answer: Out of scope, won't do, we don't have time & energy, massive complication, etc.
Fork project, start a phone/tablet/etc. version if desired & good luck 😄 

Even more detailed answer: Limited applicability for a ton of energy. DOS games are keyboard + mouse (at least the good ones). If someone wants to put in massive effort just to be able to play a few platformers on their phones (badly), it's their call.

**Reply by Torinde** [reply]
OK. Hopefully someone does that.

> DOS games are keyboard + mouse

Recent developments on the Android/iOS side go precisely in that direction - for foldables, "phone docking stations" (or just USB-C monitor+hub), Chromebook laptops, etc.:
- [Android desktop mode](https://www.androidauthority.com/try-android-16-desktop-mode-on-pixel-3565393/)
- [future ChromeOS to be based on Android](https://www.slashgear.com/1879570/chromebook-android-operating-system-chromeos-what-expect-change-means/)
- [With iPadOS 26, Apple has finally turned the iPad into a computer](https://9to5mac.com/2025/06/11/with-ipados-26-apple-has-finally-turned-the-ipad-into-a-computer/)
- and of course Android/iOS devices already support connecting to external keyboard and mouse

**Reply by FeralChild64** [reply]
@Torinde You are free to fork DOSBox Staging and add support for these platform by yourself. To me, testing my changes on Linux, macOS, and Windows (and sometimes writing quite a lot of platform-specific code) is already enough of a hassle.

**Reply by johnnovak** [reply]
100% what @FeralChild64 said. I won't go near mobile development in this lifetime, especially not if it involves DOSBox.

**Comment by Grounded0** [comment]
We definitely don't view any Android or Libretro core forks negatively since we're not doing them.

**Comment by Torinde** [comment]
What's the situation with this? Are those contact attempts described somewhere?
- [Magic Dosbox](http://magicbox.imejl.sk/) ([Play Store](https://play.google.com/store/apps/details?id=bruenor.magicbox.free)) - GPL violator: modified GPL2 sources not public and does not respond to requests

I see [this](https://magicbox.imejl.sk/gpl-faq/):
> Any user who has obtained a copy of Magic DOSBox from Google Play Store can email us and we’ll provide the source code(in next few business days). We dont support pirated downloads out of google play in any way. Please provide your trackable order number to the email for the donated version or any proof that free version was downloaded from google play.

Is this how GPL works? Or they should publish the code publicly without such "ask and you shall receive" step?

**Reply by Grounded0** [reply]
> Is this how GPL works? Or they should publish the code publicly without such "ask and you shall receive" step?

That's a GPL violation as CodeWeavers readily provides source code for CrossOver which is a commercial open source software (good luck building it though).

I know the politics behind this that the DOSBox developers aren't willing to pursue Magic Dosbox in any way over this so there isn't really a case for going after it.

**Reply by Torinde** [reply]
OK, and what does it mean "does not respond to requests"?
a) does not respond to request to publish publicly the code
b) does not respond to request by "Any user who has obtained a copy of Magic DOSBox from Google Play Store" to get the code

If a), but not b), then I would find it strange that no user has requested and got the code and then published it/created a fork/etc.

**Reply by johnnovak** [reply]
Technically, I think you are free to provide the sources on 2 million postage stamps or something sent via mail on request only, and you're still complying with the GPL 😆 Plus I think you can ask money to cover the printing and shipping fees, etc. So yeah, there are many ways around it to be _really_ difficult if you want to 😆 

Or simply send the sources with zero documentation, yeah.

**Reply by dreamer** [reply]
> Is this how GPL works? Or they should publish the code publicly without such "ask and you shall receive" step?

Yes, this is how GPL works. You are obliged to send source code on request from users that you distributed/sold the project to. And you are obliged to do it only on request, you don't need to publish it publically. And yes, that's how some companies use GPL and it's perfectly OK.

However, in practice it's much easier to just publish source code and avoid handling user requests. It's just easier and in >99.99% cases how GPL source code is distributed.

> Technically, I think you are free to provide the sources on 2 million postage stamp (...)

Nope, that would not be right - GPL is still legal license text, not a "gotcha" script ; obfuscating code this way would count as license violation. Same if code was distributed but users would be not allowed to build it (e.g. because of some proprietary tools).

> GPL violator: modified GPL2 sources not public and does not respond to requests

I did write this and yeah, Magic Dosbox is still license violator. If they distributed only paid version then the way they operate would be obtuse but legal. But they DO distribute Magic Dosbox Lite, which is free-to-use - therefore the source code should be freely available. BTW, in the past (long time ago) I did send a request for their source code for Lite version and was ignored.

**Reply by dreamer** [reply]
> Are those contact attempts described somewhere?

I just checked and I still have that mail in my "Sent" dir I sent it in September 2020 to magicbox@imejl.sk (this email used to be somewhere on their page, now I cannot find any email address there), with the "proof" of using Lite version (screenshots). They never responded.

**Reply by FeralChild64** [reply]
@dreamer Some ideas:
1. Report them to Google, https://support.google.com/legal/answer/3110420?sjid=11618543117085943870-EU
2. Report them to FSF, https://www.gnu.org/licenses/gpl-violation.html
